extensions/ql-vscode/src/databases/db-config.ts

```diff
@@ -0,0 +1,29 @@
+// Contains models for the data we want to store in the database config
+
+export interface DbConfig {
+  remote: RemoteDbConfig;
+}
+
+export interface RemoteDbConfig {
+  repositoryLists: RemoteRepositoryList[];
+  owners: string[];
+  repositories: string[];
+}
+
+export interface RemoteRepositoryList {
+  name: string;
+  repositories: string[];
+}
+
+export function cloneDbConfig(config: DbConfig): DbConfig {
+  return {
+    remote: {
+      repositoryLists: config.remote.repositoryLists.map((list) => ({
+        name: list.name,
+        repositories: list.repositories.slice(),
```

Comment:
Out of interest, what's the reason for doing `list.repositories.slice()` here but doing `[...config.remote.owners]` later on?

Reply:
No good reason - updated to use spread everywhere. In theory I think `slice` might be faster in our case but I don't think it matters here, so any would do.